src/deadline/client/cli/_deadline_cli.py

```diff
@@ -34,50 +25,68 @@
     "DEBUG",
 ]  # Log Levels AWS Deadline Cloud Allows
 
-# Set the default log level based on the setting, must do here so we can pass into the click option
-_SETTING_LOG_LEVEL = get_setting("settings.log_level").upper()
-_DEFAULT_LOG_LEVEL = get_setting_default("settings.log_level")
-_CLI_DEFAULT_LOG_LEVEL = _DEFAULT_LOG_LEVEL
-if _SETTING_LOG_LEVEL not in _DEADLINE_LOG_LEVELS:
-    logger.warning(
-        f"Log Level '{_SETTING_LOG_LEVEL}' not in {_DEADLINE_LOG_LEVELS}. Defaulting to {_DEFAULT_LOG_LEVEL}"
-    )
-else:
-    _CLI_DEFAULT_LOG_LEVEL = _SETTING_LOG_LEVEL
+
+def _get_default_log_level() -> str:
+    """
+    Get the default log level from the config file.
+    """
+    # Set the default log level based on the setting, must do here so we can pass into the click option
+    _SETTING_LOG_LEVEL = get_setting("settings.log_level").upper()
+    _DEFAULT_LOG_LEVEL = get_setting_default("settings.log_level")
+    _CLI_DEFAULT_LOG_LEVEL = _DEFAULT_LOG_LEVEL
+    if _SETTING_LOG_LEVEL not in _DEADLINE_LOG_LEVELS:
+        logger.warning(
+            f"Log Level '{_SETTING_LOG_LEVEL}' not in {_DEADLINE_LOG_LEVELS}. Defaulting to {_DEFAULT_LOG_LEVEL}"
+        )
+    else:
+        _CLI_DEFAULT_LOG_LEVEL = _SETTING_LOG_LEVEL
+    return _CLI_DEFAULT_LOG_LEVEL
+
+
+class ContextTrackingCommand(click.Command):
+    """
+    Adds the current CLI command name to User Agent headers in boto requests
+    """
+
+    def invoke(self, ctx: click.Context):
+        # This is a global variable used to modify User Agent header in the default boto config
+        session_context["cli-command-name"] = ctx.command_path.replace(" ", ".")
+        return super().invoke(ctx)
+
+
+class ContextTrackingGroup(click.Group):
+    """
+    Adds the current CLI command name to User Agent headers in boto requests
+    """
+
+    # Special value documented in Click to make this group class the default
+    group_class = type
+
+    # Special value documented in Click to make this command class the default
+    command_class = ContextTrackingCommand
 
 
-@click.group(context_settings=CONTEXT_SETTINGS)
+@click.group(cls=ContextTrackingGroup, context_settings=CONTEXT_SETTINGS)
 @click.version_option(version=version, prog_name="deadline")
 @click.option(
     "--log-level",
     type=click.Choice(_DEADLINE_LOG_LEVELS, case_sensitive=False),
-    default=_CLI_DEFAULT_LOG_LEVEL,
+    default=None,
```

Comment:
I think this affects the help output of the command. What does that look like before/after?